### PR TITLE
Add docs on custom admin

### PR DESCRIPTION
**Pull request by idanl21**
After a lot of issues opened, I decided to test it on my own and write some guide about it.
We also need still to figure out why users fail to change the admin password on running clusters.

### CODE REVIEW

**Comment by CodeGlitcher** [comment]
Hi there,

I think this is a bad solution to changing the admin password.
(The reason why noone changes it, is because its to hard)
Because now i have to put my password in some online tool and add the hash to my code or use a local tool to create the hash..

Would it not be better for the operator to use the admin cert that is generated to communicate with the cluster.
And let the operator create a random admin user (or let users now define a admin user with OpensearchUser object)

I was going to make a issue but then i saw this pr. 
So I thought I would post my thoughts here because we are running into the same problem.
We want a good way to change the admin password, but it is all a bit hacky.

a follow up issue we have is that each time there is a change in the security config files.
The operator restarts opensearch-cluster-securityconfig-update. But this removes alle the users created by the operator.
Because the security tooling removes all users that are not in the internalusers.yml. 
After the job the operator does not seem to try to recreate the OpenSearchUsers unil a restart

(When we deployed opensearch ourself with pure docker 
we detected if the security index was already present and used
/securityadmin.sh -f $1 to apply indivudiual files. 
and we skipped internaluser.yml to preserve our users)

**Comment by idanl21** [comment]
> Hi there,
> 
> I think this is a bad solution to changing the admin password. (The reason why noone changes it, is because its to hard) Because now i have to put my password in some online tool and add the hash to my code or use a local tool to create the hash..
> 
> Would it not be better for the operator to use the admin cert that is generated to communicate with the cluster. And let the operator create a random admin user (or let users now define a admin user with OpensearchUser object)
> 
> I was going to make a issue but then i saw this pr. So I thought I would post my thoughts here because we are running into the same problem. We want a good way to change the admin password, but current setup is a bit hacky.
> 
> We also tryed to create the admin user in a seperate job, but this also causes some startup problems, because when the operator cant create a user it simple stops creating it, rather then creating it when we have created the admin user.
> 
> a follow up issue we have is that each time there is a change in the security config files. The operator restarts opensearch-cluster-securityconfig-update. But this removes alle the users created by the operator. Because the security tooling removes all users that are not in the internalusers.yml. After the job the operator does not seem to try to recreate the OpenSearchUsers unil a restart
> 
> (When we deployed opensearch ourself with pure docker we detected if the security index was already present and used /securityadmin.sh -f $1 to apply indivudiual files. and we skipped internaluser.yml to preserve our users)

So a lot of customers asked as for that feature, people want to know the admin users and to be able to configure it?
@swoehrl-mw @prudhvigodithi, what is your opinion ?

**Comment by swoehrl-mw** [comment]
@CodeGlitcher You make several points:

> Would it not be better for the operator to use the admin cert that is generated to communicate with the cluster.

That would be a way but also makes it more complex so we'll have to think about this more deeply.

> And let the operator create a random admin user (or let users now define a admin user with OpensearchUser object)

At the moment the operator follows how opensearch itself does it and that means default credentials and the user has to change them, which is possible (if a little cumbersone) the way @idanl21 wants to document it in this PR.

> We also tryed to create the admin user in a seperate job [...]

In terms of configuration and securityconfig the operator is all-in, you are not supposed to manage parts separately as that would make everything more complex.

> a follow up issue we have is that each time there is a change in the security config files [...]

This is a known and open issue: #254 and we will have to think on how we handle that.

File: docs/userguide/main.md [hunk 1]
```diff
@@ -477,6 +477,55 @@ spec:
         secretName: secret-name
 ```
 
+
+## Custom Admin User
+
+In order to create your cluster with other than default `admin:admin` user you will have to add walk through those steps:
```
Comment (swoehrl-mw) [inline docs/userguide/main.md:483]:
```suggestion
In order to create your cluster with an adminuser different from the default `admin:admin` you will have to walk through the following steps: 
```

File: docs/userguide/main.md [hunk 2]
```diff
@@ -477,6 +477,55 @@ spec:
...
+  name: admin-credentials-secret
+type: Opaque
+data:
+  # admin
+  username: YWRtaW4=
+  # admin123
+  password: YWRtaW4xMjM=
+```
+
+Second you will have to create your own `securityconfig-secret` secret (take a look about the at opensearch-operator/examples/securityconfig-secret.yaml for an example).
```
Comment (swoehrl-mw) [inline docs/userguide/main.md:498]:
```suggestion
Second you will have to create your own `securityconfig-secret` secret (take a look at `opensearch-operator/examples/securityconfig-secret.yaml` for an example).
```

File: opensearch-operator/examples/opensearch-cluster-custom-admin-user.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+apiVersion: opensearch.opster.io/v1
```
Comment (swoehrl-mw) [inline opensearch-operator/examples/opensearch-cluster-custom-admin-user.yaml:1]: Please add a comment as first line explaining what this file is an example for and when to use it

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]

**Pull request merged**